src/jshint.js

```diff
@@ -2496,7 +2496,7 @@ var JSHINT = (function() {
     if (left) {
       if (left.type === "(identifier)") {
         if (left.value.match(/^[A-Z]([A-Z0-9_$]*[a-z][A-Za-z0-9_$]*)?$/)) {
-          if ("Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
+          if ("Array Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
```

Comment:
That seems more appropriate. I was basing this fix off of a previous commit for a similar issue: https://github.com/jshint/jshint/commit/2fecdea9fbc8c66e659b4be7035f03bab16b855e